### PR TITLE
fix: Polygon delete

### PR DESCRIPTION
**Pull request by alainajaffer**
Deleting the polygon should work in all cases on services and stops

### CODE REVIEW

File: site/components/map/DrawControl.tsx [hunk 1]
```diff
@@ -22,14 +35,81 @@ const DrawControl = ({
...
+                    }
+                } else if (features.length === 1 && features[0].id) {
+                    if (e.mode === "direct_select") {
+                        selectFeature(features[0].id.toString());
+                    }
+                }
+            });
+
+            map.on("draw.selectionchange", (e: { mode: DrawMode; features: PolygonFeature[] }) => {
+                deleteFeatures(e, onDelete);
```
[the remark this answers is not in the document]
Comment (alainajaffer) [reply site/components/map/DrawControl.tsx:109]: i'll ask steven

File: site/components/map/DrawControl.tsx [hunk 2]
```diff
@@ -22,14 +35,81 @@ const DrawControl = ({
...
+                    draw.deleteAll();
+                    onDelete();
+                }, 0);
+            });
+            map.on("draw.modechange", (e: { mode: DrawMode }) => {
+                const features = draw.getAll().features;
+
+                if (features.length > 1 && features[0].id) {
+                    if (e.mode === "draw_polygon") {
+                        const data = draw.getAll();
```
[the remark this answers is not in the document]
Comment (alainajaffer) [reply site/components/map/DrawControl.tsx]: fixed

File: site/components/map/DrawControl.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (alainajaffer) [reply site/components/map/DrawControl.tsx]: resolved

File: site/components/map/DrawControl.tsx [hunk 3]
```diff
@@ -22,14 +35,64 @@ const DrawControl = ({
...
+        () => {
+            return draw;
+        },
+
+        ({ map }) => {
             map.on("draw.create", onCreate);
             map.on("draw.update", onUpdate);
-            map.on("draw.delete", onDelete);
+            map.on("draw.delete", () => {
+                setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (alainajaffer) [reply site/components/map/DrawControl.tsx]: done

File: site/components/map/DrawControl.tsx [hunk 4]
```diff
@@ -22,14 +35,64 @@ const DrawControl = ({
...
+                    }
+                }
+            });
+
+            map.on("draw.selectionchange", () => {
+                const features = draw.getAll().features;
+                if (features.length === 0) {
+                    draw.changeMode("draw_polygon");
+                }
+                if (features.length > 0 && features[0].id) {
```
[the remark this answers is not in the document]
Comment (alainajaffer) [reply site/components/map/DrawControl.tsx]: done

**Review by ljones92 — APPROVED** [review]

**Pull request merged**